crates/core/tedge_agent/src/agent.rs

```diff
@@ -628,21 +629,27 @@ impl SmAgent {
 }
 
 #[cfg(test)]
-fn get_restart_command() -> Command {
+fn get_restart_operation_command_vec() -> Vec<Command> {
```

Comment:
Minor - adding type information in a function or value name is a bad habit, because your are repeating yourself and will have to maintain in sync the name and the type (say you want to remove duplicates by returning a set).

```suggestion
fn get_restart_operation_commands -> Vec<Command> {
```